pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -106,10 +104,107 @@ func (d *DefaultEvictor) addAllConstraints(logger klog.Logger, handle frameworkt
 		d.constraints = append(d.constraints, constraints...)
 	}
 	d.constraints = append(d.constraints, evictionConstraintsForMinPodAge(args.MinPodAge)...)
-	d.constraints = append(d.constraints, evictionConstraintsForIgnorePodsWithoutPDB(args.IgnorePodsWithoutPDB, handle)...)
 	return nil
 }
 
+// applyEffectivePodProtections applies all relevant protection pod policies to the plugin.
+// For each:
+// - If it's in the list, we disallow eviction (shouldEvict = false).
+// - If it's NOT in the list, we allow eviction (shouldEvict = true).
+func applyEffectivePodProtections(d *DefaultEvictor, podProtections []PodProtection, handle frameworktypes.Handle) error {
+	policyMap := make(map[PodProtection]bool)
+	for _, policy := range podProtections {
+		policyMap[policy] = true
+	}
+
+	d.constraints = append(d.constraints, evictionConstraintsForFailedBarePods(d.logger, !policyMap[FailedBarePods])...)
+	if constraints, err := evictionConstraintsForSystemCriticalPods(d.logger, !policyMap[SystemCriticalPods], d.args.PriorityThreshold, handle); err != nil {
+		return err
+	} else {
+		d.constraints = append(d.constraints, constraints...)
+	}
+	d.constraints = append(d.constraints, evictionConstraintsForLocalStoragePods(!policyMap[PodsWithLocalStorage])...)
+	d.constraints = append(d.constraints, evictionConstraintsForDaemonSetPods(!policyMap[DaemonSetPods])...)
+
+	d.constraints = append(d.constraints, evictionConstraintsForPvcPods(policyMap[PodsWithPVC])...)
+	d.constraints = append(d.constraints, evictionConstraintsForIgnorePodsWithoutPDB(policyMap[PodsWithoutPDB], handle)...)
+	return nil
+}
+
+// contains checks if a slice of PodProtectionPolicy contains a given policy.
+func contains(list []PodProtection, item PodProtection) bool {
+	for _, v := range list {
+		if v == item {
+			return true
+		}
+	}
+	return false
+}
+
+// getEffectivePodProtections determines which policies are currently active.
+// It supports both new-style (PodProtections) and legacy-style flags.
+func getEffectivePodProtections(args *DefaultEvictorArgs) []PodProtection {
+	// determine whether to use PodProtections config
+	useNewConfig := len(args.PodProtections.DefaultDisabled) > 0 || len(args.PodProtections.ExtraEnabled) > 0
+
+	if !useNewConfig {
+		// fall back to the Deprecated config
+		return legacyGetPodProtections(args)
+	}
+
+	effective := make([]PodProtection, 0)
+	effective = append(effective, DefaultPodProtections...)
+
+	for _, disabledPolicy := range args.PodProtections.DefaultDisabled {
+		// Find the index of the disabled policy in the effective slice.
+		for i, policy := range effective {
+			if policy == disabledPolicy {
+				// Remove the policy
+				effective = append(effective[:i], effective[i+1:]...)
+				break
+			}
+		}
+	}
+
+	// Add extra enabled policies if not included
+	for _, policy := range args.PodProtections.ExtraEnabled {
+		if !contains(effective, policy) {
```

Comment:
I don't think this is necessary. There is no way effective will contain anything other than DefaultPodProtections, right ? Assuming the configuration has been sanitized this could be replaced with the way simpler:

```go
effective = append(effective, args.PodProtections.ExtraEnabled...)
```